File: puzzles/day 6/py/six.one.py
from pathlib import Path
input = Path("../input.txt").read_text("utf-8")
data = input.split()
starters = [int]
print(f"array length: {len(data)}")
for x in range(len(data)):
    occurance = 0
    sequence = [input[x], input[x+1], input[x+2], input[x+3]]
    print(f"current sequence of array: {sequence}")
    for val in sequence:
        if (sequence.count(val) > 1):
            occurance += 1
    if (occurance == 0):
        starters.append(x+4)
print(starters)

File: puzzles/day 6/js/six.one.js
const fs =require('fs');
const datastream = fs.readFileSync("../input.txt",'utf-8');
const data = datastream.split("");
let starters = [];
for (let i = 0; i < data.length; i++) {
    let occurance = 0
    const sequence =[data[i],data[i+1],data[i+2],data[i+3]]
    sequence.map((val)=>{
        if(sequence.indexOf(val) !== sequence.lastIndexOf(val)){
            occurance++;
        }
    })
    if(occurance ==0){
        starters.push(i+4)
    }
}
console.log(starters[0])
